lib/managers/webhooks.js

```diff
@@ -0,0 +1,138 @@
+/**
+* @Webhookoverview Manager for the Box Webhooks Resource
+* @author ccheng
+*/
+'use strict';
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+var urlPath = require('../util/url-path');
+
+// ------------------------------------------------------------------------------
+// Private
+// ------------------------------------------------------------------------------
+
+// Base path for all webhooks endpoints
+var BASE_PATH = '/webhooks';
+
+// ------------------------------------------------------------------------------
+// Public
+// ------------------------------------------------------------------------------
+
+/**
+* Simple manager for interacting with all 'Webhooks' endpoints and actions.
+*
+* @param {BoxClient} client The Box API Client that is responsible for making calls to the API
+* @constructor
+*/
+function Webhooks(client) {
+	// Attach the client, for making API calls
+	this.client = client;
+}
+
+
+/**
+* Returns a webhook object if creation succeeds.
+*
+* API Endpoint: '/webhooks'
+* Method: POST
+*
+* @param {string} id - File or Folder ID to create webhook on
+* @param {string} type - Value of 'file' or 'folder'
+* @param {string} notificationURL - The url of your application where Box will notify you of events triggers
+* @param {string[]} triggerTypes - Array of event types that trigger notification for the target
+* @param {Function} callback - Passed the file information if it was acquired successfully
+* @returns {void}
+*/
+Webhooks.prototype.createWebhook = function(id, type, notificationURL, triggerTypes, callback) {
+	var params = {
+		body: {
+			target: {
+				id: id,
+				type: type
+			},
+			address: notificationURL,
+			triggers: triggerTypes
+		}
+	};
+
+	var apiPath = urlPath(BASE_PATH);
+	this.client.post(apiPath, params, this.client.defaultResponseHandler(callback));
+};
+
+/**
+* Returns a webhook object
+*
+* API Endpoint: '/webhooks/ID'
+* Method: GET
+*
+* @param {string} webhookID - ID of the webhook to retrieve
+* @param {Function} callback - Passed the file information if it was acquired successfully
+* @returns {void}
+*/
+Webhooks.prototype.getWebhook = function(webhookID, callback) {
```

Comment:
The `fields` query parameter is available on most (if not all) GET resources, and allows the caller to customize which pieces of information on the object are returned: https://docs.box.com/reference#fields